src/products/products-dynamodb.handler.ts

```diff
@@ -0,0 +1,99 @@
+import * as AWS from 'aws-sdk';
+
+const dynamodb = new AWS.DynamoDB.DocumentClient();
+
+export const getProductsList = async (_event: any = {}): Promise<any> => {
+  try {
+    const productsResult = await dynamodb.scan({ TableName: process.env.PRODUCTS_TABLE_NAME || '' }).promise();
+    const stockResult = await dynamodb.scan({ TableName: process.env.STOCK_TABLE_NAME || '' }).promise();
+
+    const products = productsResult.Items || [];
+    const stock = stockResult.Items || [];
+
+    const productList = products.map((product) => {
+      const stockItem = stock.find((item: any) => item.product_id === product.id);
+      return {
+        id: product.id,
+        title: product.title,
+        description: product.description,
+        price: product.price,
+        count: stockItem ? stockItem.count : 0
+      };
+    });
+
+    return {
+      statusCode: 200,
+      body: JSON.stringify(productList)
+    };
+  } catch (error: any) {
+    return {
+      statusCode: 500,
+      body: JSON.stringify({ message: 'Error fetching products: ' + error.message })
+    };
+  }
+};
+
+export const getProductsById = async (event: any = {}): Promise<any> => {
+  try {
+    const productId = event.pathParameters.productId;
+
+    const productResult = await dynamodb.get({
+      TableName: process.env.PRODUCTS_TABLE_NAME || '',
+      Key: {
+        id: productId
+      }
+    }).promise();
+
+    if (!productResult.Item) {
+      return {
+        statusCode: 404,
+        body: JSON.stringify({ message: 'Product not found' })
+      };
+    }
+
+    return {
+      statusCode: 200,
+      body: JSON.stringify(productResult.Item)
+    };
+  } catch (error) {
+    console.error('Error fetching product:', error);
+    return {
+      statusCode: 500,
+      body: JSON.stringify({ message: 'Error fetching product' })
+    };
+  }
+};
+
+export const createProduct = async (event: any = {}): Promise<any> => {
```

Comment:
the function is not fully correct. In ideal world it should also populate the stock table with a count. But it wasn't mentioned in the task, only in a optional task about transaction. So you can skip it if you want.